components/clientComponents/forms/AddressComplete/AddressComplete.tsx

```diff
@@ -0,0 +1,343 @@
+
+    const query = e.target.value;
+    const responseData = await getAddressCompleteChoices(
+      apiKey,
+      query,
+      addressObject?.country || "CAN"
+    );
+
+    //loop through the responseData and add it to the addressResultsCache
+    const newElements: AddressCompleteChoice[] = [];
+
+    for (let i = 0; i < responseData.length; i++) {
+      // Check key doesn't already exist.
+      if (
+        !addressResultCache.find((item: AddressCompleteChoice) => item.Id === responseData[i].Id)
+      ) {
+        newElements.push(responseData[i]);
+      }
+    }
+
+    if (newElements.length > 0) {
+      setAddressResultCache((prevCache) => [...prevCache, ...newElements]);
+    }
+
+    // Filter the results to avoid duplicate entry
+    const uniqueResults = responseData.filter(
+      (item: AddressCompleteChoice, index: number, self: AddressCompleteChoice[]) =>
+        index ===
+        self.findIndex((t) => toFullAddress(t) === toFullAddress(item) && item.Text !== undefined)
+    );
+
+    setChoices(
+      uniqueResults.map((item: AddressCompleteChoice) => {
+        return toFullAddress(item);
+      })
+    );
+  };
+
+  const onAddressSet = async (value: string) => {
+    if (!allow) {
+      return;
+    } // Abandon if addressComplete is disabled.
+
+    const selectedResult = addressResultCache.find(
+      (item: AddressCompleteChoice) => toFullAddress(item) === value
+    );
+    if (selectedResult === undefined) {
+      return; // Do nothing, this is not found in the AddressComplete API.
+    } else {
+      const responseData = await getSelectedAddress(
+        apiKey,
+        selectedResult.Id,
+        addressObject?.country || "CAN"
+      );
+      if (responseData) {
+        const results = responseData;
+        setAddressObject(results);
+      }
+    }
+  };
+
+  const setAddressData = (key: string, value: string) => {
+    let baseAddressObject = {};
+
+    if (addressObject === null) {
+      baseAddressObject = {
+        unitNumber: "",
+        civicNumber: "",
+        streetName: "",
+        city: "",
+        province: "",
+        postalCode: "",
+        country: "",
+      };
+    } else {
+      baseAddressObject = addressObject;
+    }
+
+    for (const internalKey in baseAddressObject) {
+      if (key === internalKey) {
+        const newAddressObject = { ...baseAddressObject, [key]: value };
+        setAddressObject(newAddressObject as AddressElements);
+      }
+    }
+  };
+
+  const countryChoices = countries.all?.map((country) => {
+    return country[i18n.language as Language];
+  });
+
+  const setCountry = (countryText: string) => {
+    // Get the country from the countries.all object.
+    const country = countries.all.find(
+      (country) => country[i18n.language as Language] === countryText
+    );
+    if (country) {
+      // set the country in the address object to the ID of the country.
+      setAddressData("country", country.id);
+    }
+  };
+
+  return (
+    <>
```

Comment:
For a followup PR. 
A fieldset requires a legend. Swapping the Description out for a Legend won't work because it may or may not be present. One idea is to have some screen reader only visible text generically describing the fields e.g. "Fill in the Address fields" or something like that.
Here is the WCAG bit if anyone is curious: https://www.w3.org/TR/WCAG20-TECHS/H71.html